test/easylistTestCases.js

```diff
@@ -0,0 +1,60 @@
+const easylistTestCases = [
+    // blocking by address parts
+    // /banner/*/img
+    {'url': 'http://example/banner/foo/img', 'block': true, 'options': {}},
+    {'url': 'http://example/banner/foo/bar/img?param', 'block': true, 'options': {}},
+    {'url': 'http://example/banner//img/foo', 'block': true, 'options': {}},
+    {'url': 'http://example/banner/img', 'block': false, 'options': {}},
+    {'url': 'http://example/banner/foo/imgraph', 'block': false, 'options': {}},
+    {'url': 'http://example/banner/foo/img.gif', 'block': false, 'options': {}},
+
+    // blocking by domain
+    // ||ads.example.com^
+    {'url': 'http://ads.example.com/foo.gif', 'block': true, 'options': {}},
+    {'url': 'http://server1.ads.example.com/foo.gif', 'block': true, 'options': {}},
+    {'url': 'http://ads.example.com:8000/', 'block': true, 'options': {}},
+    {'url': 'http://ads.example.com.ua/foo.gif', 'block': false, 'options': {}},
+    {'url': 'http://example2.com/foo.gif', 'block': false, 'options': {}},
+    {'url': 'http://example.info/redirect/http://example.com/', 'block': false, 'options': {}},
+
+    // blocking options
+    // ||ads.somesite.com^$script,image,domain=example.com|~foo.example.info
```

Comment:
Can you add a note here specifying how `domain=example.com|~foo.example.info` modifies matches? I want for someone to be able to walk thru these test cases and understand them without having prior knowledge.

Reply:
Note is great. Thanks